### PR TITLE
Allow Bundle to run in a target namespace

### LINKED ISSUE #51: Bundle Lib needs to expose an option to deploy the APB in the same namespace as the target namespace
## Feature

The bundle lib at times will need to deploy the apb into the same namespace as the target of the apb. 

## Current Behavior

There is no way to currently tell bundle-lib to not create the transient namespace.

### PR DESCRIPTION
**Pull request by shawn-hurley**
fixes #51 

~~depends on #65~~

### CODE REVIEW

File: bundle/bind.go [hunk 1]
```diff
@@ -37,24 +44,49 @@ func (e *executor) Bind(
 
 	go func() {
 		e.actionStarted()
-		executionContext, err := e.executeApb(
-			"bind", instance.Spec, instance.Context, parameters)
+		// Create namespace name that will be used to generate a name.
+		ns := fmt.Sprintf(fmt.Sprintf("%s-%.4s-", instance.Spec.FQName, bindAction))
```
Comment (jmrodri) [inline bundle/bind.go]: Why do we `fmt.Sprintf` twice?

File: bundle/bind.go [hunk 2]
```diff
@@ -37,24 +44,49 @@ func (e *executor) Bind(
...
+			BundleName: pn,
+			Targets:    targets,
+			Metadata:   labels,
+			Action:     bindAction,
+			Image:      instance.Spec.Image,
+		}
+
+		serviceAccount, namespace, err := runtime.Provider.CreateSandbox(pn, ns, targets, clusterConfig.SandboxRole, labels)
+		ec.Account = serviceAccount
+		ec.Location = namespace
```
Comment (jmrodri) [inline bundle/bind.go]: The order of this initialization feels weird. We create the `ExecutionContext` (partially), then get 2 values, which we add to the context. Feels like we should get the `serviceAccount` and the `namespace` first, handle the error. If it's okay, then create the ExecutionContext.

File: bundle/deprovision.go [hunk 3]
```diff
@@ -42,24 +49,48 @@ func (e *executor) Deprovision(instance *ServiceInstance) <-chan StatusMessage {
 			e.actionFinishedWithError(errors.New("No image field found on instance.Spec"))
 			return
 		}
+		// Create namespace name that will be used to generate a name.
+		ns := fmt.Sprintf(fmt.Sprintf("%s-%.4s-", instance.Spec.FQName, deprovisionAction))
```
Comment (jmrodri) [inline bundle/deprovision.go]: Double `fmt.Sprintf`?

File: bundle/deprovision.go [hunk 4]
```diff
@@ -42,24 +49,48 @@ func (e *executor) Deprovision(instance *ServiceInstance) <-chan StatusMessage {
...
+			Targets:    targets,
+			Metadata:   labels,
+			Action:     deprovisionAction,
+			Image:      instance.Spec.Image,
+		}
 
-		// Might need to change up this interface to feed in instance ids
-		executionContext, err := e.executeApb("deprovision", instance.Spec,
-			instance.Context, instance.Parameters)
+		serviceAccount, namespace, err := runtime.Provider.CreateSandbox(pn, ns, targets, clusterConfig.SandboxRole, labels)
```
Comment (jmrodri) [inline bundle/deprovision.go]: Same comment as above.

**Review by jmrodri — CHANGES_REQUESTED** [review]
I think we should create the context after we get the service account and namespace in all of the places.

File: bundle/bind.go (hunk 1 above)
Comment (dymurray) [reply bundle/bind.go]: +1 seems redundant

File: bundle/bind.go [hunk 5]
```diff
@@ -37,24 +44,49 @@ func (e *executor) Bind(
...
+			ec.BundleName,
+			ec.Location,
+			ec.Targets,
 			clusterConfig.Namespace,
 			clusterConfig.KeepNamespace,
 			clusterConfig.KeepNamespaceOnError,
 		)
 		if err != nil {
-			log.Errorf("Problem executing apb [%s] bind", executionContext.PodName)
+			log.Errorf("Problem executing apb [%s] bind", ec.BundleName)
```
Comment (dymurray) [inline bundle/bind.go]: `Problem executing bundle [%s] bind` for consistency

**Comment by dymurray** [comment]
The approach makes sense to me. Just a couple nits and questions Jesus mentioned but other than that looks good.

**Comment by jmrodri** [comment]
@shawn-hurley needs a rebase

**Review by jmrodri — APPROVED** [review]
ACK

**Review by dymurray — APPROVED** [review]

**Pull request merged**